packages/core/src/services/commits/promptCache.ts

```diff
@@ -0,0 +1,98 @@
+import { hash } from 'crypto'
+
+import { Config, Conversation } from '@latitude-data/compiler'
+
+import { ChainStepResponse, StreamType, type Workspace } from '../../browser'
+import { cache } from '../../cache'
+
+class DocumentCache {
+  private workspace: Workspace
+  private config: Config
+  private conversation: Conversation
+
+  constructor({
+    workspace,
+    config,
+    conversation,
+  }: {
+    workspace: Workspace
+    config: Config
+    conversation: Conversation
+  }) {
+    this.workspace = workspace
+    this.config = config
+    this.conversation = conversation
+  }
+
+  private async getCache() {
+    return await cache()
+  }
+
+  private generateKey() {
+    const k = hash(
+      'sha256',
+      `${JSON.stringify(this.conversation)}:${JSON.stringify(this.config)}`,
+    )
+
+    return `workspace:${this.workspace.id}:prompt:${k}`
+  }
+
+  private shouldCache(): boolean {
+    const temp = parseFloat(this.config.temperature as any)
+    return isNaN(temp) || temp === 0
+  }
+
+  async get(): Promise<ChainStepResponse<StreamType> | undefined> {
```

Comment:
Micro ocd. But put private functions under public methods

Reply:
I'm removing the class